lib/NormalModuleFactory.js

```diff
@@ -849,6 +849,25 @@ ${err2.stack}`;
 								err.message += `
 ${hints.join("\n\n")}`;
 							}
+
+							// Check if the extension is missing a leading dot (e.g. "js" instead of ".js")
+							let appendResolveExtensionsHint = false;
+							const specifiedExtensions = Array.from(
+								resolver.options.extensions
+							);
+							const expectedExtensions = specifiedExtensions.map(extension => {
+								if (extension.match(/^[^.]/)) {
```

Comment:
Hoist this RegExp literal into a variable. This reduces memory usage and GC.

Reply:
See https://github.com/webpack/webpack/blob/b604d7818241ebef02e4785cf94b10a9ec0f4ebf/lib/SourceMapDevToolPlugin.js#L50-L56 for example